payplug.php

```diff
@@ -100,22 +103,102 @@ private function upgrade()
 					// If is defined
 					else
 						$os_payment = defined('_PS_OS_PAYMENT_');
-					Configuration::updateValue('PAYPLUG_ORDER_STATE_PAID', (int)$os_payment);
+
+					Payplug::updateConfiguration('PAYPLUG_ORDER_STATE_PAID', (int)$os_payment);
 				}
 			}
+			// Add test status && add hook
+			if ($version === false || version_compare($version, '0.9.7', '<'))
+			{
+				Configuration::deleteByName('PAYPLUG_ORDER_STATE_REFUND');
+
+				Payplug::updateConfiguration('PAYPLUG_SANDBOX', '0');
+
+				$install = new InstallPayplug();
+				$install->createOrderState();
+
+				if (version_compare(_PS_VERSION_, '1.5', '<'))
+					$this->registerHook('header');
+				else
+					$this->registerHook('displayHeader');
+
+				$install->installPayplugLock();
+
+			}
 
 			// Upgrade in DataBase the new version
-			Configuration::updateValue($cfg_name, $this->version);
+			Payplug::updateConfiguration($cfg_name, $this->version, true);
 		}
 	}
 
+	/**
+	 * Update configuration
+	 * @var string  Configuration key
+	 * @var mixed   Configuration value
+	 * @var boolean If is global configuration
+	 */
+	public static function updateConfiguration($key, $value, $global = false)
+	{
+		if (version_compare(_PS_VERSION_, '1.5', '>=') && $global)
+			Configuration::updateGlobalValue($key, $value);
+		else
+			Configuration::updateValue($key, $value);
+	}
+
+	/**
+	 * Get configuration
+	 * @var string   Configuration key
+	 * @var boolean  If is global configuration
+	 * @return mixed Value of key, or false if not exists
+	 */
+	public static function getConfiguration($key, $global = false)
+	{
+		if (version_compare(_PS_VERSION_, '1.5', '>=') && $global)
+			$value = Configuration::getGlobalValue($key);
+		else
+			$value = Configuration::get($key);
+
+		return $value;
+	}
+
+	/**
+	 * Get Order State configuration
+	 * @param  string  $state_name State name
+	 * @return integer             Value
+	 */
+	public static function getOsConfiguration($state_name)
+	{
+		$key = 'PAYPLUG_ORDER_STATE_'.Tools::strtoupper($state_name);
+
+		if (self::getConfiguration('PAYPLUG_SANDBOX'))
+			$key .= '_TEST';
+
+		return self::getConfiguration($key, false);
+	}
+
 	public function install()
 	{
 		if (version_compare(_PS_VERSION_, '1.4', '<') || !parent::install() || !$this->registerHook('payment') || !$this->registerHook('paymentReturn'))
 			return false;
+
+		// add hook for 1.6
+		if (version_compare(_PS_VERSION_, '1.5', '<'))
+		{
+			if (!$this->registerHook('header'))
+				return false;
```

Comment:
```php
	if (!$this->registerHook('header'))
		return false;
```
will work on PS 1.4, but 1.5 and 1.6 too. You do not need to check the version for a displayHeader case.